.travis.yml

```diff
@@ -0,0 +1,15 @@
+language: perl
+perl:
+    - "5.20"
+    - "5.18"
+    - "5.16"
+    - "5.14"
+before_install:
+    - "cpanm --installdeps ."
+    - "cpanm Date::Utility" 
+    - "cpanm Test::Mock::Redis"
+    - "cpanm DBD::Mock"
+    - "cpanm Moose"
+    - "cpanm JSON"
+script:
+  perl Makefile.PL && make test
```

Comment:
please make sure all the modules are installed and the environment variables set to run the optional tests on travis. Otherwise they are never run.